pkg/services/ngalert/api/api_prometheus.go

```diff
@@ -71,9 +80,17 @@ func (srv PrometheusSrv) RouteGetRuleStatuses(c *models.ReqContext) response.Res
 		namespaceUIDs = append(namespaceUIDs, k)
 	}
 
+	dashboardUID := c.Query("dashboard_uid")
+	panelID, err := getPanelIDFromRequest(c.Req)
```

Comment:
alternatively, you could use [QueryInt64](https://gowalker.org/gopkg.in/macaron.v1#Context_QueryInt64)

Reply:
I chose not to use `QueryInt64` as it doesn't return an error if the query string contains an invalid integer.